### PR TITLE
Bug/2 card pickup

### PR DESCRIPTION
**Pull request by austinbrianrogers**
Added a "focus" flag to the mouse picker to know if it's currently interacting with a UI element, as to not click on other elements.

Currently only the dragndrop classes use this feature.

### CODE REVIEW

File: kitten/mouse picking/MousePicker.h [hunk 1]
```diff
@@ -9,6 +9,7 @@ namespace MousePicker
 {
 	bool rayHits(const kitten::Ray& p_ray, const kitten::ClickableBox* p_clickable, double* p_hitOutput);	
 	bool uiHit(const kitten::ClickableFrame* p_clickable, float p_mouseX, float p_mouseY);
+	static bool m_isFocused = false;
```
Comment (calmackenzie) [inline kitten/mouse picking/MousePicker.h]: Could probably put this static bool in `DragNDrop` as it is only set and accessed there

File: UI/DragNDrop.h [hunk 2]
```diff
@@ -5,7 +5,7 @@
 class DragNDrop : public kitten::ClickableUI
 {
 protected:
-	bool m_isDragging=false, m_backToOrigin= false;
+	bool m_isDragging, m_backToOrigin= false;
```
Comment (calmackenzie) [inline UI/DragNDrop.h]: Can you add `m_isDragging(false)` in the constructor of `DragNDrop.cpp` ?

File: UI/DragNDrop.cpp [hunk 3]
```diff
@@ -32,8 +37,8 @@ void DragNDrop::update()
 	if (m_isDragging)
 	{
 		m_attachedObject->getTransform().place2D(
-			input::InputManager::getInstance()->getMouseXPos() - 5,
-			input::InputManager::getInstance()->getMouseYOpenGLPos() - getTransform().getScale2D().y + 5
+			input::InputManager::getInstance()->getMouseXPos() - 25,
+			input::InputManager::getInstance()->getMouseYOpenGLPos() - getTransform().getScale2D().y + 25
```
Comment (calmackenzie) [inline UI/DragNDrop.cpp]: I'm not sure what this change does, but if it is a visual change that is fine -- just adding comment so you are aware of it

Comment (austinbrianrogers) [reply UI/DragNDrop.cpp]: Visual Change

File: kitten/mouse picking/MousePicker.h (hunk 1 above)
Comment (austinbrianrogers) [reply kitten/mouse picking/MousePicker.h]: changed

File: UI/DragNDrop.h (hunk 2 above)
Comment (austinbrianrogers) [reply UI/DragNDrop.h]: done

File: UI/DragNDrop.cpp [hunk 4]
```diff
@@ -3,27 +3,28 @@
 
 #define TIME_TO_GO_BACK_TO_ORIGIN 0.025
```
Comment (calmackenzie) [inline UI/DragNDrop.cpp]: did you forget to define the static bool in this file? I do not see it

Comment (austinbrianrogers) [reply UI/DragNDrop.cpp]: added to constructor - though it worked fine without this

Comment (calmackenzie) [reply UI/DragNDrop.cpp]: I meant putting `bool DragNDrop::m_focused = false;` at the top of the file, not in the constructor -- does the game build ?

Comment (calmackenzie) [reply UI/DragNDrop.cpp]: Also don't want this in the constructor as a new card could be created while one is being dragged and screw things up

**Pull request merged**